### PR TITLE
inline format!() args from rustc_middle up to and including rustc_codegen_llvm (3)

### PR DESCRIPTION
**Pull request by matthiaskrgr**
r? @WaffleLapkin

### CODE REVIEW

File: compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs [hunk 1]
```diff
@@ -2467,81 +2462,81 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
...
-                );
-                span.push_span_label(
-                    interior_span,
-                    format!("has type `{}` which {}", target_ty, trait_explanation),
-                );
-                if let Some(scope_span) = scope_span {
-                    let scope_span = source_map.end_point(scope_span);
+        let mut explain_yield = |interior_span: Span,
+                                 yield_span: Span,
+                                 scope_span: Option<Span>| {
```
[the remark this answers is not in the document]
Comment (matthiaskrgr) [reply compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs:2496]: Hmm, my uneducated guess is that rustfmt simply bailed out from formatting the block previously and the format!() change somehow gave it a little confidence boost...

[the remark this answers is not in the document]
Comment (WaffleLapkin) [reply compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs:2496]: Makes sense

**Review by WaffleLapkin — APPROVED** [review]

**Pull request merged**